crates/dekaf/src/session.rs

```diff
@@ -1318,6 +1350,30 @@ impl Session {
         }
     }
 
+    async fn fetch_collection_partitions(
```

Comment:
Now that I'm reading this again, it seems like this function could instead just return the `Collection`s themselves, and let the caller get the number of partitions from them if needed.

Reply:
I started out that way and the amount of duplication felt too smelly so I extracted it... but yeah, I think I went a tiny bit too far by throwing away the `Collection`s in favor of their `partitions.len()`. Will update this to return the corresponding Collections